### PR TITLE
simple-http-server: 0.6.1 -> 0.6.2

### PR DESCRIPTION
**Pull request by Mephistophiles**
###### Description of changes

Bump to 0.6.2: https://github.com/TheWaWaR/simple-http-server/releases/tag/v0.6.2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/simple-http-server/default.nix [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
     rev = "v${version}";
-    sha256 = "01a129i1ph3m8k6zkdcqnnkqbhlqpk7qvvdsz2i2kas54csbgsww";
+    sha256 = "sha256-ndLFN9FZZA+zsb+bjZ3gMvQJqo6I92erGOQ44H+/LCg=";
   };
 
-  cargoSha256 = "050avk6wff8v1dlsfvxwvldmmgfakdxmhglv2bhvc2f3q8cf1d5d";
+  cargoSha256 = "sha256-3wbdfuj4jikLsq5GQXsB3Tsbx1KJPg744VtZrhln4u4=";
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ pkg-config perl ];
```
Comment (Mephistophiles) [inline pkgs/servers/simple-http-server/default.nix:16]: **note:** The new version requires perl for openssl-sys (rust):

```
Caused by:
  process didn't exit successfully: `/build/source/target/release/build/openssl-sys-e280345a7f6b4c22/build-script-main` (exit status: 101)
  --- stdout
  cargo:rustc-cfg=const_fn
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_OPENSSL_NO_VENDOR
  X86_64_UNKNOWN_LINUX_GNU_OPENSSL_NO_VENDOR unset
  cargo:rerun-if-env-changed=OPENSSL_NO_VENDOR
  OPENSSL_NO_VENDOR unset
  CC_x86_64-unknown-linux-gnu = Some("/nix/store/6mx62f4mdv5y1awi7y2mp23a76gg0ba7-gcc-wrapper-11.3.0/bin/cc")
  CFLAGS_x86_64-unknown-linux-gnu = None
  CFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running "perl" "./Configure" "--prefix=/build/source/target/x86_64-unknown-linux-gnu/release/build/openssl-sys-7290fd8aa5dc2e9f/out/openssl-build/install" "--openssldir=/usr/local/ssl" "no-dso" "no-shared" "no-ssl3" "no-unit-test" "no-comp" "no-zlib" "no-zlib-dynamic" "no-md2" "no-rc5" "no-weak-ssl-ciphers" "no-camellia" "no-idea" "no-seed" "linux-x86_64" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64"

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /build/simple-http-server-0.6.2-vendor.tar.gz/openssl-src/src/lib.rs:488:39
```

**Review by winterqt — CHANGES_REQUESTED** [review]
Please follow the commit name guidelines from [the manual](https://nixos.org/manual/nixpkgs/stable/#submitting-changes-making-patches), i.e. `simple-http-server: 0.6.1 -> 0.6.2`.

**Comment by Mephistophiles** [comment]
@winterqt Oh, sorry about this. Fixed! Thanks for review!

**Review by winterqt — CHANGES_REQUESTED** [review]
So this release added a vendored OpenSSL dependency, which ideally we'd remove.

Here's how you'd do it:
1. Update `Cargo.toml` to not use the `vendored` feature on the `openssl` crate, and generate a patch (`git diff`) for the change
2. Update the `Cargo.lock` file (`cargo update`)
3. Copy both the patch and the updated `Cargo.lock` to this package's folder
4. Add the patch to `patches`, copy the `Cargo.lock` in `postPatch`, and set `cargoLock.lockFile` to the `Cargo.lock` file

You can look around the repo to see examples of how this is done (like [here](https://github.com/NixOS/nixpkgs/blob/14a9b2760461c50607c8241607125cb2ddf99106/pkgs/servers/binserve/default.nix)), and I'll of course let you know if you did something incorrectly (so don't be afraid to make mistakes). If you need any clarifications on these steps, feel free to let me know.

**Comment by Mephistophiles** [comment]
@winterqt thanks for this detailed review! It was awesome!

:heavy_check_mark: added `Cargo.lock`
:heavy_check_mark: patched `Cargo.toml`
:heavy_check_mark: removed perl from `Cargo.toml` - it no longer needed

**Review by winterqt — CHANGES_REQUESTED** [review]
Let's fix Darwin support.

```diff
-  buildInputs = if stdenv.isDarwin then [ Security ] else [ openssl ];
+  buildInputs = [ openssl ] ++ lib.optional stdenv.isDarwin Security;
```

**Comment by Mephistophiles** [comment]
@winterqt Fixed. All checks have passed, thanks for your review!

**Review by winterqt — APPROVED** [review]

**Pull request merged**